src/hapi/pipelines/database/food_security.py

```diff
@@ -23,138 +23,189 @@ def __init__(
-                        ipc_type = _get_ipc_type_code_from_data(
-                            ipc_type_from_data=values[ipc_type_column][
-                                admin_code
-                            ][irow]
-                        )
-                        (
-                            time_period_start,
-                            time_period_end,
-                        ) = _get_time_period(
-                            month_range=values[reference_period_months_column][
-                                admin_code
-                            ][irow],
-                            year=values[reference_period_year_column][
-                                admin_code
-                            ][irow],
-                        )
-                        # Total population required to calculate fraction in phase
-                        population_total = int(
-                            values[population_in_phase_columns["all"]][
-                                admin_code
-                            ][irow]
+                else:
+                    adminone_name = row["Level 1"]
+                    subnational_level = admin_level
+                    if not adminone_name and subnational_level == "admintwo":
+                        subnational_level = "adminone"
+                        adminone_name = row["Area"]
+                    if countryiso3 in food_sec_config["adm2_only"]:
+                        if subnational_level == "adminone":
+                            adminone_name = None
+                            adminone = None
+                            subnational_level = "admintwo"
+                        else:
+                            continue
+                    elif adminone_name:
+                        full_adm1name = f"{countryiso3}|{adminone_name}"
+                        if any(
+                            x in adminone_name.lower()
+                            for x in food_sec_config["adm_ignore_patterns"]
+                        ):
+                            add_message(
+                                warnings,
+                                dataset_name,
+                                f"Admin 1: ignoring {full_adm1name}",
+                            )
+                            continue
+                        adminone, exact = self._adminone.get_pcode(
+                            countryiso3, adminone_name
                         )
-                        # Sum the population in each row by type and date to aggregate admin 1.5 to admin 1
-                        dict_of_lists_add(
-                            population_totals,
-                            (
-                                ipc_type,
-                                time_period_start,
-                                time_period_end,
-                            ),
-                            population_total,
+                    else:
+                        continue
```

Comment:
Should this be recorded as an error - are there population numbers in these rows typically? Any idea what they could be referring to if there is no admin 1?

Reply:
I have added more error handling in the refactoring